### PR TITLE
make pytest report py3 issues

### LINKED ISSUE #77: sqlmeta class has no attribute soClass
https://travis-ci.org/sqlobject/sqlobject/jobs/50224733#L220

### PR DESCRIPTION
**Pull request by lukmdo**
use pytest in addition to flake8 to hunt for py3 problems

resolves #77

### CODE REVIEW

**Comment by lukmdo** [comment]
[currently travis](https://travis-ci.org/sqlobject/sqlobject/jobs/50783719) reports only
> AttributeError: type object 'sqlmeta' has no attribute 'soClass'

File: sqlobject/compat.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+def with_metaclass(meta, *bases):
```
Comment (lukmdo) [inline sqlobject/compat.py]: not sure why six is not part of py3 migration ?

**Comment by phdru** [comment]
Looks good. I think it can goes in. Thanks, Lukasz!

**Comment by lukmdo** [comment]
:beers: @phdru six dependency could be better then [compat.py](https://github.com/sqlobject/sqlobject/pull/80/files#diff-294b6e0cd3cabd19e6b87d984176f149)

**Comment by phdru** [comment]
Let's wait and hear if @sigmavirus24 and @drnlm have something to say about it.

Personally I have no experience with six. As a matter of fact I haven't switched to Py3 at all.

**Comment by sigmavirus24** [comment]
For the exactly one function we need, which is extraordinarily simple, I don't think we need a six dependency. That said, since you've taken `with_metaclass` from [six](https://bitbucket.org/gutworth/six/src/986089f3d04e468458cc9f5b9b3be26fb3d75ade/six.py?at=default#cl-790) I would greatly appreciate it if you would give proper attribution to the six authors and contributors in the compat module.

**Comment by lukmdo** [comment]
thanks updated - hoped six would help more with other porting issues. But @sigmavirus24 makes perfect sense.

guess best to follow 3to2 standard when possible (example: using `range` not aliased `xrange`)

**Pull request merged**